system-test/client-side-metrics.ts

```diff
@@ -373,98 +603,7 @@ describe('Bigtable/ClientSideMetrics', () => {
-              delete secondRequest.firstResponseLatency;
-              delete secondRequest.applicationLatencies;
-              delete secondRequest.metricsCollectorData.appProfileId;
-              assert.deepStrictEqual(secondRequest, {
-                status: '0',
-                streaming: 'true',
-                client_name: 'nodejs-bigtable',
-                metricsCollectorData: {
-                  instanceId: 'emulator-test-instance',
-                  cluster: 'fake-cluster3',
-                  zone: 'us-west1-c',
-                  method: 'Bigtable.ReadRows',
-                  table: 'my-table',
-                },
-                projectId,
-                retryCount: 0,
-              });
-              // We would expect these parameters to be different every time so delete
-              // them from the comparison after checking they exist.
-              const thirdRequest = this.requestsHandled[2] as any;
-              assert(thirdRequest.attemptLatency);
-              assert(thirdRequest.serverLatency);
-              delete thirdRequest.attemptLatency;
-              delete thirdRequest.serverLatency;
-              delete thirdRequest.metricsCollectorData.appProfileId;
-              assert.deepStrictEqual(thirdRequest, {
-                connectivityErrorCount: 0,
-                streaming: 'true',
-                status: '0',
-                client_name: 'nodejs-bigtable',
-                metricsCollectorData: {
-                  instanceId: 'emulator-test-instance',
-                  table: 'my-table2',
-                  cluster: 'fake-cluster3',
-                  zone: 'us-west1-c',
-                  method: 'Bigtable.ReadRows',
-                },
-                projectId,
-              });
-              const fourthRequest = this.requestsHandled[3] as any;
-              // We would expect these parameters to be different every time so delete
-              // them from the comparison after checking they exist.
-              assert(fourthRequest.operationLatency);
-              assert(fourthRequest.firstResponseLatency);
-              assert(fourthRequest.applicationLatencies);
-              delete fourthRequest.operationLatency;
-              delete fourthRequest.firstResponseLatency;
-              delete fourthRequest.applicationLatencies;
-              delete fourthRequest.metricsCollectorData.appProfileId;
-              assert.deepStrictEqual(fourthRequest, {
-                status: '0',
-                streaming: 'true',
-                client_name: 'nodejs-bigtable',
-                metricsCollectorData: {
-                  instanceId: 'emulator-test-instance',
-                  cluster: 'fake-cluster3',
-                  zone: 'us-west1-c',
-                  method: 'Bigtable.ReadRows',
-                  table: 'my-table2',
-                },
-                projectId,
-                retryCount: 0,
-              });
```

Comment:
The above code has been moved out into a function and the check has been abstracted so that it can vary for each test. This is because the check is different for application latencies.